src/WebJobs.Script/Workers/SharedMemoryDataTransfer/MemoryMappedFileAccessorUnix.cs

```diff
@@ -127,6 +136,64 @@ public override void Delete(string mapName, MemoryMappedFile mmf)
             }
         }
 
+        /// <summary>
+        /// Checks if a list of directories is specified in AppSettings to create <see cref="MemoryMappedFile"/>.
+        /// If one is specified, returns that list. Otherwise returns the default list.
+        /// </summary>
+        /// <returns>List of paths of directories where <see cref="MemoryMappedFile"/> are allowed to be created.</returns>
+        private List<string> GetAllowedDirectories()
+        {
+            string envVal = _environment.GetEnvironmentVariable(RpcWorkerConstants.FunctionsUnixSharedMemoryDirectories);
+            if (string.IsNullOrEmpty(envVal))
+            {
+                return SharedMemoryConstants.TempDirs;
+            }
+
+            return envVal.Split(',').ToList();
+        }
+
+        /// <summary>
+        /// From a list of allowed directories where <see cref="MemoryMappedFile"/> can be created, it will return
+        /// a list of those that are valid (i.e. exist, or have been successfully created).
+        /// </summary>
+        /// <returns>List of paths of directories where <see cref="MemoryMappedFile"/> can be created.</returns>
+        private List<string> GetValidDirectories()
+        {
+            List<string> allowedDirectories = GetAllowedDirectories();
+            List<string> validDirectories = new List<string>();
+
+            foreach (string directory in allowedDirectories)
+            {
+                string path = Path.Combine(directory, SharedMemoryConstants.TempDirSuffix);
```

Comment:
Should this "TempDirSuffix" also be an environment variable?

Reply:
Umm, I think this is fine. That will be too many configurable pieces and I don't see a use for that. The user can set the main path and inside that we create a subdirectory by this name.